### ISSUE TITLE
core_user_get_users web service doesn't always return customfields

### ISSUE THREAD

**Issue by davidpesce** [iomad/iomad#2106]
We upgraded from IOMAD 3.9.x to 4.1.15 and noticed that the core_user_get_users (core Moodle) function does not always return custom fields in the response. We have "optional profiles" for all tenants and only one of them returns the customfields.

System details:
IOMAD 4.1.15 (Build: 20241209)
PHP 7.4.x
Apache 2.4

**Comment by davidpesce** [comment]
A quick update on this. If I assign the web service authorized user to a specific company, the customfields are returned for users within that company only. I would think that being an authorized user outside of the company scope, would return all users with the customfields within it.

**Comment by turf212** [comment]
If the authorised user is tied to a company then they should only get their companies profile field information - so it would be the same as if they were editing users in that tenant. If they also have the capability block/iomad_company_admin:allcompany_user_profiles then they will get to see all of the profile fields for all of the companies.

**Comment by davidpesce** [comment]
Also of note, if I create a new user (outside of the company scope, ie, in Moodle), when using that user scope via the API, it only returns the first company created in IOMAD (even though it's not assigned to a company). If I then assign the user to a company, the API only returns customfields for that specific company.

**Comment by turf212** [comment]
Hmm - should probably not be returning any company profile fields in that instance.  Still - if you add the capability mentioned - do you get everything?

**Comment by davidpesce** [comment]
No, the user has that permission and it still only returns what is in their own company.
